straxen/plugins/peaks/peak_nearest_triggering.py

```diff
@@ -0,0 +1,144 @@
+import numpy as np
+import numba
+import strax
+import straxen
+from .peak_ambience import _quick_assign
+from ..events import Events
+
+export, __all__ = strax.exporter()
+
+
+@export
+class PeakNearestTriggering(Events):
+    """Time difference and properties of the nearest triggering peaks in the left and right
+    direction of peaks."""
+
+    __version__ = "0.0.0"
+    depends_on = ("peak_basics", "peak_proximity")
+    provides = "peak_nearest_triggering"
+    data_kind = "peaks"
+    save_when = strax.SaveWhen.EXPLICIT
+
+    shadow_time_window_backward = straxen.URLConfig(
+        default=int(1e9),
+        type=int,
+        track=True,
+        help="Search for peaks casting time & position shadow in this time window [ns]",
+    )
+
+    only_trigger_min_area = straxen.URLConfig(
+        default=False,
+        type=bool,
+        track=True,
+        help="Whether only require the triggering peak to have area larger than trigger_min_area",
+    )
+
+    def infer_dtype(self):
+        dtype = []
+        common_descr = "of the nearest triggering peak on the"
+        for direction in ["left", "right"]:
+            dtype += [
+                (
+                    (f"time difference {common_descr} {direction} [ns]", f"{direction}_dtime"),
+                    np.int64,
+                ),
+                ((f"time {common_descr} {direction} [ns]", f"{direction}_time"), np.int64),
+                ((f"endtime {common_descr} {direction} [ns]", f"{direction}_endtime"), np.int64),
+                (
+                    (f"center_time {common_descr} {direction} [ns]", f"{direction}_center_time"),
+                    np.int64,
+                ),
+                ((f"type {common_descr} {direction}", f"{direction}_type"), np.int8),
+                ((f"n_competing {common_descr} {direction}", f"{direction}_n_competing"), np.int32),
+                ((f"area {common_descr} {direction} [PE]", f"{direction}_area"), np.float32),
+            ]
+        dtype += strax.time_fields
+        return dtype
+
+    def get_window_size(self):
+        # This method is required by the OverlapWindowPlugin class
+        return 10 * self.shadow_time_window_backward
+
+    def compute(self, peaks):
+        argsort = np.argsort(peaks["center_time"], kind="mergesort")
+        _peaks = np.sort(peaks, order="center_time")
+        result = np.zeros(len(peaks), self.dtype)
+        _quick_assign(argsort, result, self.compute_triggering(peaks, _peaks))
```

Comment:
Just for my own understanding, this `_quick_assign` just places the peaks back in the original order, correct?